Content.Shared/Nyanotrasen/Item/PseudoItem/SharedPseudoItemSystem.cs

```diff
@@ -0,0 +1,158 @@
+        StorageComponent? storage = null)
+    {
+        if (!Resolve(storageUid, ref storage))
+            return false;
+
+        if (!CheckItemFits((toInsert, component), (storageUid, storage)))
+            return false;
+
+        var itemComp = new ItemComponent
+            { Size = component.Size, Shape = component.Shape, StoredOffset = component.StoredOffset };
+        AddComp(toInsert, itemComp);
+        _item.VisualsChanged(toInsert);
+
+        _tag.TryAddTag(toInsert, PreventTag);
+
+        if (!_storage.Insert(storageUid, toInsert, out _, null, storage))
+        {
+            component.Active = false;
+            RemComp<ItemComponent>(toInsert);
+            return false;
+        }
+
+        component.Active = true;
+        return true;
+    }
+
+    private void OnEntRemoved(EntityUid uid, PseudoItemComponent component, EntGotRemovedFromContainerMessage args)
+    {
+        if (!component.Active)
+            return;
+
+        RemComp<ItemComponent>(uid);
+        component.Active = false;
+    }
+
+    private void OnGettingPickedUpAttempt(EntityUid uid, PseudoItemComponent component,
+        GettingPickedUpAttemptEvent args)
+    {
+        if (args.User == args.Item)
+            return;
+
+        Transform(uid).AttachToGridOrMap();
+        args.Cancel();
+    }
+
+    private void OnDropAttempt(EntityUid uid, PseudoItemComponent component, DropAttemptEvent args)
+    {
+        if (component.Active)
+            args.Cancel();
+    }
+
+    private void OnInsertAttempt(EntityUid uid, PseudoItemComponent component,
+        ContainerGettingInsertedAttemptEvent args)
+    {
+        if (!component.Active)
+            return;
+        // This hopefully shouldn't trigger, but this is a failsafe just in case so we dont bluespace them cats
+        args.Cancel();
+    }
+
+    // Prevents moving within the bag :)
+    private void OnInteractAttempt(EntityUid uid, PseudoItemComponent component, InteractionAttemptEvent args)
+    {
+        if (args.Uid == args.Target && component.Active)
+            args.Cancel();
+    }
+
+    private void OnDoAfter(EntityUid uid, PseudoItemComponent component, DoAfterEvent args)
+    {
+        if (args.Handled || args.Cancelled || args.Args.Used == null)
+            return;
+
+        args.Handled = TryInsert(args.Args.Used.Value, uid, component);
+    }
+
+    protected void StartInsertDoAfter(EntityUid inserter, EntityUid toInsert, EntityUid storageEntity,
+        PseudoItemComponent? pseudoItem = null)
+    {
+        if (!Resolve(toInsert, ref pseudoItem))
+            return;
+
+        var ev = new PseudoItemInsertDoAfterEvent();
+        var args = new DoAfterArgs(EntityManager, inserter, 5f, ev, toInsert, toInsert, storageEntity)
+        {
+            BreakOnTargetMove = true,
+            BreakOnUserMove = true,
+            NeedHand = true
+        };
+
+        _doAfter.TryStartDoAfter(args);
+    }
```

Comment:
We'll need to add a check for OnAttackAttempt, so that Felinids can't attack people near/holding the bag.

Reply:
Fixed. The client will still predict the attack and play the sound/animation in the case of melee hits, but nothing will actually register